src/preferences/dialog/dlgprefmixer.cpp

```diff
@@ -1017,58 +1119,74 @@ void DlgPrefMixer::validate_levels() {
-    EffectSlotPointer pEffectSlot(m_pEffectMainEQ);
-    if (!pEffectSlot.isNull()) {
-        auto pParameterSlot = pEffectSlot->getEffectParameterSlot(
-                EffectManifestParameter::ParameterType::Knob, i);
-
-        if (pParameterSlot->isLoaded()) {
-            QSlider* slider = m_mainEQSliders[i];
-            // Calculate parameter value from relative slider position
-            double paramValue = static_cast<double>(value - slider->minimum()) /
-                    static_cast<double>(slider->maximum() - slider->minimum());
-            pParameterSlot->setParameter(paramValue);
-            slider->setValue(static_cast<int>(value * 100));
-
-            QLabel* valueLabel = m_mainEQValues[i];
-            QString valueText = QString::number(value);
-            valueLabel->setText(valueText);
-
-            m_pConfig->set(ConfigKey(kConfigGroup,
-                                   kMainEQParameterKey + QString::number(i + 1)),
-                    ConfigValue(valueText));
+void DlgPrefMixer::storeMainEQ() {
+    // store effect
+    const EffectManifestPointer pManifest =
+            m_pBackendManager->getManifestFromUniqueId(
+                    comboBoxMainEq->currentData().toString());
+
+    m_pConfig->set(ConfigKey(kMixerProfile, kEffectGroupForMaster),
+            ConfigValue(pManifest ? pManifest->uniqueId() : ""));
+
+    // clear all previous parameter values so we have no residues
+    const QList<ConfigKey> mixerKeys = m_pConfig->getKeysWithGroup(kMixerProfile);
+    for (const auto& key : mixerKeys) {
+        if (key.item.contains(kMainEQParameterKey)) {
+            m_pConfig->remove(key);
         }
     }
+
+    // store current parameters
+    for (QSlider* pSlider : std::as_const(m_mainEQSliders)) {
+        int paramIndex = pSlider->property("index").toInt();
+        double dispValue = static_cast<double>(pSlider->value()) / 100;
+        // TODO store this in effects.xml
+        m_pConfig->set(ConfigKey(kMixerProfile,
+                               kMainEQParameterKey + QString::number(paramIndex + 1)),
+                ConfigValue(QString::number(dispValue)));
+    }
 }
 
-const QList<EffectManifestPointer> DlgPrefMixer::getFilteredManifests(
-        EffectManifestFilterFnc filterFunc) const {
+const QList<EffectManifestPointer> DlgPrefMixer::getDeckEqManifests() const {
     const QList<EffectManifestPointer> allManifests =
             m_pBackendManager->getManifests();
-    if (filterFunc == nullptr) {
-        return allManifests;
+    QList<EffectManifestPointer> eqs;
+    QList<EffectManifestPointer> nonEqs;
+    // Add EQs first, then append non-EQs
+    for (const auto& pManifest : allManifests) {
+        if (isMixingEQ(pManifest.data())) {
+            eqs.append(pManifest);
+        } else if (!m_eqEffectsOnly) {
+            nonEqs.append(pManifest);
+        }
     }
+    // Append non-EQs. No-op if 'EQs only' is unchecked.
+    eqs.append(nonEqs);
+    return eqs;
```

Comment:
same thing, just faster and easier to read IMO. if you need the order to stay the same as in `allManifests` (which I don't think you do tbh), use `stable_partition` instead. 
```suggestion
    auto nonEqsStartIt = std::partition(allManifests.begin(), allManifests.end(), [](const auto& pManifest) {
        return isMixingEQ(pManifest.data());
    });
    if (m_eqEffectsOnly) {
        allManifests.erase(nonEqsStartIt, allManifests.end());
    }
    return allManifests;
```

Reply:
Thanks for another std magic trick!

Had to remove `const` from `allManifests`, and yes, the [lexigraphically sorted list](https://github.com/mixxxdj/mixxx/blob/5b32026fa7cea38291b577367907887e8ad6af90/src/effects/backends/effectmanifest.cpp#L17-L29) was fine so I used `stable_partition`.

I was wondering why Balance was listed as Stereo Balance so I also changed the combobox items to use the `displayName`. And I added a separator between the EQs and non-EQs.